### PR TITLE
rgw: fix segfault in RevokeThread during its shutdown procedure.

### PR DESCRIPTION
**Pull request by rzarzynski**
Fixes: http://tracker.ceph.com/issues/19831
Signed-off-by: Radoslaw Zarzynski <rzarzynski@mirantis.com>

### CODE REVIEW

File: src/rgw/rgw_keystone.h [hunk 1]
```diff
@@ -224,7 +224,7 @@ class TokenCache {
     friend class TokenCache;
     typedef RGWPostHTTPData RGWGetRevokedTokens;
 
-    CephContext * const cct;
+    CephContext* const cct;
```
Comment (rzarzynski) [inline src/rgw/rgw_keystone.h]: In the future we might want to switch to `boost::intrusive_ptr<CephContext>`.

Comment (cbodley) [inline src/rgw/rgw_keystone.h]: why not now? it should be a drop-in replacement here

**Comment by rzarzynski** [comment]
jenkins retest this please

**Comment by rzarzynski** [comment]
jenkins retest this please

**Comment by yuriw** [comment]
ready for merge http://pulpito.ceph.com/yuriw-2017-07-21_19:21:22-rgw-wip-yuri-testing_2017_7_22-distro-basic-smithi/

**Comment by cbodley** [comment]
jenkins test this please

**Comment by cbodley** [comment]
@rzarzynski may need a rebase for the `src/test/cli/crushtool/build.t` failure

**Comment by rzarzynski** [comment]
Rebasing.

**Comment by rzarzynski** [comment]
jenkins retest this please (`run-cli-tests (Failed)`)

**Pull request merged**